src/neuroconv/datainterfaces/ecephys/neuralynx/neuralynxdatainterface.py

```diff
@@ -1,106 +1,71 @@
-        if line:
-            if line[0] == "-":
-                key, val = line[1:].split(" ", 1)
-                header_dict[key] = val
-    return header_dict
-
-
-def get_metadata(folder_path: FolderPathType) -> dict:
-    """
-    Parse the header of one of the .ncs files to get the session start time (without
-    timezone) and the session_id.
-
-    Parameters
-    ----------
-    folder_path: str or Path
-
-    Returns
-    -------
-    dict
-    """
-    folder_path = Path(folder_path)
-    csc_files = sorted(folder_path.glob("*.[nN]cs"))
-    file_path = csc_files[0]
-    with file_path.open(encoding="latin1") as file:
-        raw_header = file.read(1024)
-    header = parse_header(raw_header)
-    if header.get("FileVersion") == "3.4":
-        return dict(
-            session_start_time=parser.parse(header["TimeCreated"]),
-            session_id=header["SessionUUID"],
-        )
-    if header.get("FileVersion", "").startswith("3.3") or header["CheetahRev"].startswith("5.4"):
-        open_line = raw_header.split("\n")[2]
-        spliced_line = open_line[24:35] + open_line[-13:]
-        return dict(session_start_time=parser.parse(spliced_line, dayfirst=False))
-
-
-def get_filtering(channel_path: FolderPathType) -> str:
-    """
-    Get the filtering metadata from an .nsc file.
-
-    Parameters
-    ----------
-    channel_path: str or Path
-        Filepath for an .nsc file
-
-    Returns
-    -------
-    str:
-        json dump of filter parameters. Uses the mu character, which may cause problems
-        for downstream things that expect ASCII.
-    """
-    channel_path = Path(channel_path)
-    with open(channel_path, "r", encoding="latin1") as file:
-        raw_header = file.read(1024)
-    header = parse_header(raw_header)
-
-    return json.dumps(
-        {key: val.strip(" ") for key, val in header.items() if key.lower().startswith("dsp")}, ensure_ascii=False
-    )
-
-
 class NeuralynxRecordingInterface(BaseRecordingExtractorInterface):
     """Primary data interface for converting Neuralynx data. Uses
     :py:class:`~spikeinterface.extractors.NeuralynxRecordingExtractor`."""
 
     def __init__(self, folder_path: FolderPathType, verbose: bool = True):
 
-        self.nsc_files = natsorted([str(x) for x in Path(folder_path).iterdir() if ".ncs" in x.suffixes])
+        self.ncs_files = natsorted([str(x) for x in Path(folder_path).iterdir() if ".ncs" in x.suffixes])
 
         super().__init__(folder_path=folder_path, verbose=verbose)
+        neo_reader = self.recording_extractor.neo_reader
         self.recording_extractor = self.recording_extractor.select_segments(segment_indices=0)
-
+        # preserve neo_reader attribute for SelectSegmentRecording
+        self.recording_extractor.neo_reader = neo_reader
         self.add_recording_extractor_properties()
 
     def add_recording_extractor_properties(self):
+        filtering = get_filtering_multi_channel(self.recording_extractor.neo_reader)
 
         try:
```

Comment:
Can you remove the try structure? it is no longer needed here.

Reply:
Done.